code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -68,6 +68,15 @@
 	internal_organs += new /obj/item/organ/alien/eggsac
 	..()
 
+/mob/living/carbon/alien/humanoid/royal/queen/proc/check_hostile()
+	SIGNAL_HANDLER
+	if(is_station_level(src.z)) //we don't want the hostile environment if the xenos aren't actually on station
+		SSshuttle.registerHostileEnvironment(src) //aliens delay shuttle
+		addtimer(CALLBACK(src, .proc/game_end), 30 MINUTES) //time until shuttle is freed/called
```

Comment:
What if queen leaves z level, comeback, leaves, does it a few times. This will queue timer calls, you should check somehow if this timer call hasn't been already queued.

Reply:
Talked about this, but I'll say it again. The hostile environment gets cleared each time her z level changes to a z level that isn't the station.

Reply:
Wait I completely misread. AAAAAAAA

Reply:
Resolved.